raiden/tests/integration/test_provisioning.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+import pytest
+import gevent
+from gevent.pool import Pool
+from ethereum import slogging
+
+from raiden.tests.utils.blockchain import wait_until_block
+
+log = slogging.getLogger(__name__)
+
+
+@pytest.mark.parametrize('number_of_nodes', [6])
+@pytest.mark.parametrize('channels_per_node', [0])
+@pytest.mark.parametrize('cached_genesis', [False])
+@pytest.mark.parametrize('register_tokens', [True, False])
+def test_participant_selection(
+    raiden_network,
+    tokens_addresses,
+    settle_timeout,
+    reveal_timeout,
+    blockchain_type
+):
+    token_address = tokens_addresses[0]
+
+    # adjust timeouts
+    for app in raiden_network:
+        app.raiden.config['reveal_timeout'] = reveal_timeout
+        app.raiden.config['settle_timeout'] = settle_timeout
+
+    # connect the first node (will register the token if necessary)
+    raiden_network[0].raiden.api.connect_token_network(token_address, 100)
+
+    # connect the other nodes
+    pool = Pool()
+    calls = []
+    for app in raiden_network[1:]:
+        api = app.raiden.api
+        calls.append((api.connect_token_network, token_address, 100))
+
+    pool.imap(lambda x: x[0](*x[1:]), calls).join()
+
+    # wait some blocks to let the network connect
+    wait_blocks = 15
+    if blockchain_type == 'geth':
```

Comment:
there is no need to special case 'geth', [wait_until_block](https://github.com/raiden-network/raiden/blob/master/raiden/tests/utils/blockchain.py#L46-L51) uses the chain proxy, and the tester implementation [will mine a block at the time](https://github.com/raiden-network/raiden/blob/master/raiden/tests/utils/tester_client.py#L226-L228)